controllers/articleController.js

```diff
@@ -165,11 +172,92 @@ const reactToArticle = async (req, res) => {
   }
 };
 
+/**
+ * bookmarks an article
+ * @param {object} req The request body which contain the article's slug as param.
+ * @param {object} res The response body.
+ * @returns {object} res.
+ */
+
+const bookmarkArticle = async (req, res) => {
+  const { slug } = req.params;
+  const { userId } = req;
+  try {
+    const existingArticle = await helper.findArticle(slug);
+    if (!existingArticle) return res.status(404).send({ error: 'Article Not found!' });
+    const bookmarked = await helper.bookmarkArticle(userId, slug);
+    bookmarked.article = existingArticle.dataValues;
+
+    res.status(200).send({ message: 'Article bookmarked successfully', data: bookmarked });
+  } catch (error) {
+    res.status(500).send({ error: error.message });
+  }
+};
+
+/**
+ * bookmarks an article
+ * @param {object} req The request body which contain the article's slug as param.
+ * @param {object} res The response body.
+ * @returns {object} res.
+ */
+
+const deleteBookmark = async (req, res) => {
+  const { slug } = req.params;
+  const { userId } = req;
+  try {
+    const existingArticle = await helper.findArticle(slug);
+    if (!existingArticle) return res.status(404).send({ error: 'Article Not found!' });
+    const unBookmark = await Bookmark.destroy({ where: { userId, articleSlug: slug } });
```

Comment:
Hi Afeez, nice work. Small variable name suggestion; I think we could make unBookmark a bit more descriptive. Maybe we could use 'notBookmarked' or something like that. What do you think?

Reply:
That's ok. I'll see to it